src/app/core/services/code.service.ts

```diff
@@ -31,7 +31,32 @@ export class CodeService {
 
   getStructure = async (folder: string): Promise<any> => {
     const structure: Structure = (await firstValueFrom(this.http.get(`./assets/talks/${ folder }/structure.json`)) as Structure);
-    this.structure.next(structure);
+    
+    //Create a copy structure that will be finally passed
+    var tempstruct: Structure ={ ORDER: [], STYLE: [] };
+
+    //Traverse through the object
+    for(let i in structure){  
```

Comment:
`structure` is not iterable. `structure.ORDER` is ...

This should looks something like this ...

```typescript
const tempStructure: Structure = { ORDER: [], STYLE: structure.STYLE };

structure.ORDER.forEach((key: string) => {
  if (structure[key].hasOwnProperty('visibility') === false || structure[key].visibility === true) {
    tempStructure.ORDER.push(key);
    tempStructure[key] = structure[key];
  }
});

this.structure.next(tempStructure);
```

Reply:
The structure has an `ORDER` array that are the keys that identify the slide objects. Style simply carries over.

Reply:
> The structure has an `ORDER` array that are the keys that identify the slide objects. Style simply carries over.

Yes I identified that, hence I added the logic to remove the necessary key from the `ORDER` array.

> `structure` is not iterable. `structure.ORDER` is 

Yet, it seems to work. Where the loop variable `i` traverses through the keys in the `structure` object.


Thanks for sharing the code snippet. Happy to implement the same.
However, I am curious as to what is the original solution missing? I'm sure the `tempstruct` variable generated in both cases will be the exact same.

I admit your suggested approach is simpler and uses less variables, but would like to know if that is the only advantage.

Reply:
Appreciate the response. Will make the changes immediately.

Reply:
![image](https://github.com/user-attachments/assets/d6381d20-851c-4ce9-96b1-4d09a710d9a2)

looks like I'll need to type set `structure[key]`

Reply:
I tried with the `any` keyword, and it seems to work:

```js
const objValue: any = structure[key];
      if (objValue.hasOwnProperty('visibility') === false || objValue.visibility === true) {
        tempStructure.ORDER.push(key);
        tempStructure[key] = structure[key];
      }
```

but then I found this declaration from `talk.component.ts`

```js
    const page: StructureType = (structure[key] as StructureType);
```
The latter sets a more specific type, hence I wished to use it. But that means I will have to import `StructureType` and I will have to declare the `visibility` property within it. Not sure if that is ideal.

P.S. I'm fully new to TypeScript, hence the desire to know the right approach.

Reply:
Modifying the `StructureType` is the right approach. I've tried to maintain the code correctly (typed) ... use `visible?: boolean`, then it's optional.

Reply:
Yes, I had done this on a previous commit. Will add the commit now.